frame/contracts/src/wasm/runtime.rs

```diff
@@ -824,14 +813,7 @@ impl<'a, E: Ext + 'a> Runtime<'a, E> {
 	) -> Result<ReturnCode, TrapReason> {
 		let charged = self.charge_gas(RuntimeCosts::GetStorage(self.ext.max_value_size()))?;
 		let key = self.read_sandbox_memory(memory, key_ptr, key_type.len::<E::T>()?)?;
-		let outcome = match key_type {
-			KeyType::Fix => self.ext.get_storage(
-				&FixSizedKey::try_from(key).map_err(|_| Error::<E::T>::DecodingFailed)?,
-			),
-			KeyType::Variable(_) => self.ext.get_storage_transparent(
-				&VarSizedKey::<E::T>::try_from(key).map_err(|_| Error::<E::T>::DecodingFailed)?,
-			),
-		};
+		let outcome = self.ext.get_storage(&key_type.decode(key)?);
```

Comment:
I think we should get rid of `KeyType`, too. We can implement `fn from_fixed(...)` and `fn from_variable(..)` on `Key`. The impl block should be in runtime.rs, though.

Reply:
mmm if we do that then we would have to lift these instructions to the caller 

```rust
let key = self.read_sandbox_memory(memory, key_ptr, key_type.len::<E::T>()?)?;
let key = key_type.decode(key)?
```
so that we can call the function like this

```rust
ctx.get_storage(memory, key.from_fixed(key_ptr), out_ptr, out_len_ptr)?;
// or 
ctx.get_storage(memory, key.from_variable(key_ptr), out_ptr, out_len_ptr)?;
```

but we don't want to do that before calling charge_gas first, so not sure how to go about it.

Reply:
ah nevermind I missed the tricky 
```rust
pub use crate::{
	address::{AddressGenerator, DefaultAddressGenerator},
	exec::{Frame, VarSizedKey as StorageKey},
```

Reply:
Yes. Didn't think about this. So I guess we keep the `KeyType` then.